### ISSUE TITLE
Für Kontrollen und Wiederholungsaufnahmen müssen die Daten der Baum- und Landmarkenplots in qgis kartographisch und im Aufnahmeformular diese und alle anderen aufgenommenen Daten dargestellt und verändert werden können.

### ISSUE THREAD

**Issue by b-lack** [b-lack/lfb-regeneration_wildlife_impact_monitoring#21]

**Comment by twiebke** [comment]
Baum- und Landmarkenplot dienen dem eindeutigen Wiederfinden des Stichprobenpunktes

### Baum- und Landmarkenplots sowie Transekt
Der Baumplot und Landmarkenplot soll kartographisch mit Symbolen und den Attributen dargestellt werden:
- [ ] die Bäume sollen dabei als Kreise mit den Baumfarben (siehe Styledatei Baumarten)
- [ ] die Durchmesser der Bäume definieren die Durchmesser für die Kreise - geeigneten Überzeichnungsfaktor finden
- [ ] Transekt aus Azimut mit 22 m Länge bis Baumplot 2, Azimut als Label
- [ ] Baumplot 2 als Punkt 22 m in Transektazi von SP=Baumplot1
- [ ] Baumart, Azimut und Distanz sollen als Textlabel am Baum stehen

### Dynamik
- [ ] Die kartographische Darstellung soll dynamisch an die vorhandenen Daten und die Eingaben gebunden sein 

### QGIS Geomety Generator
Ich empfehle den QGIS Geometry Generator: https://docs.qgis.org/3.28/en/docs/user_manual/style_library/symbol_selector.html#the-geometry-generator

- https://gitlab.com/GIS-projects/qgis-geometry-generator-examples
- https://gis.stackexchange.com/questions/384769/how-to-create-a-geometry-generator-symbol-using-pyqgis
- https://www.statsmapsnpix.com/2023/03/geometry-generators-in-qgis.html
- geht wohl auch von Raster: https://hannes.enjoys.it/blog/2019/09/dynamic-elevation-profile-lines-as-qgis-geometry-generator/

Video:
- https://www.youtube.com/watch?v=0YxjJ-9zIJ0

**Comment by b-lack** [comment]
Das ist doch schon eine Super Hilfe.

Ich finde, das hier trifft es am Besten: https://gis.stackexchange.com/questions/430685/applying-different-style-to-each-sub-part-of-multi-part-geometry-using-qgis

Ich habe bisher aber noch nicht gesehen, ob und wie eine dynamische Liste (Baumplot/Landmarken) als Quelle eingebunden wurde. Das muss ich testen!

**Comment by b-lack** [comment]
Habe ich jetzt mal Hardcoded ausprobiert. Es ist möglich, aber sehr zeitraubend. Würde ich entsprechend ganz nach hinten schieben.

![image](https://github.com/b-lack/lfb-regeneration_wildlife_impact_monitoring/assets/11278402/2f0e5203-a656-4a7e-aafb-96c8c73e2e0d)

**Comment by twiebke** [comment]
Verstehe ich. 
Zeitlich wichtiger ist auch ersteinmal , dass die Daten sichtbar und veränderbar sind. Die können die Punkte über die Bäume auch ersteinmal anhand der textuellen Beschreibung wiederfinden. Auf Dauer hilft es schon sehr das auch zu visualisieren.

**Comment by b-lack** [comment]
@twiebke Ich probiere jetzt schon Tage mit dem Expression-Builder das gewünschte Layout zu erzeugen. Ich finde jedoch keinen Hinweis im Netz, das es funktionieren könnte. Vielleicht weißt du mehr:

- Dynamisch erzeugte Geometrien (Bäume, Transekt, ) können meines Wissensstandes keine Labels haben.
- Dynamisch erzeugte Geometrien können sich nicht in Farbe und Form unterscheiden.

![image](https://github.com/b-lack/qgis-fim-plugin/assets/11278402/e402dd75-f972-4596-a134-e207b9fea347)
Hier im Bild dynamisch erzeugte Bäume in blau, Transekt und Landmarken in rot.

**Comment by twiebke** [comment]
Soweit ich weiß gibt es für die kartographische Darstellung grundsätzlich die Herausforderung, dass das im EPSG 4326 doof ist weil auf grad umgerechnet werden müsste und Verzerrungen auftraten. 
Überlegenswert wäre auf EPS25833 zu wechseln, weiß aber nicht ob das mit dem GNSS gut geht.

Die dynamisch erzeugten Objekte wären mir gar nicht so wichtig, das wäre schön. Wichtig ist, dass die in den gespeicherten Daten beschriebenen Objekte gezeichnet werden. Sonst ist ein Wiederfinden zur Wiederholungsinventur oder für Kontrollen erschwert. Schön ist auch, wenn der Aufnahmetrupp nach der Speicherung das Resultat kartographisch überprüfen kann.

**Comment by b-lack** [comment]
@twiebke Habe die relevante Funktion eine Sekunde nachdem ich den Kommentagr geschrieben habe gefunden:

![image](https://github.com/b-lack/qgis-fim-plugin/assets/11278402/babf8046-e179-4f97-8939-6d485a4480f9)
Bäume aktuell alle noch braun aber mit Label

Gezoomed:
![image](https://github.com/b-lack/qgis-fim-plugin/assets/11278402/be257263-cccf-4085-a2f4-a4ab6a79edd7)


Fehlt hier was wichtiges?

**Comment by twiebke** [comment]
Cool.
Sieht ersteinmal nicht so aus als ob was fehlt.

**Comment by b-lack** [comment]
Die Baumarten-Farben sind jetzt auch da:
![image](https://github.com/b-lack/qgis-fim-plugin/assets/11278402/7a3f29d8-9bc8-4c20-bd3f-e8059cb7b30b)

Mich wundert allerding, dass in der Datei https://github.com/b-lack/qgis-fim-plugin/blob/0fb0c6ef1e3f25eaa8d88343a8f87a87c130f892/lfb_regeneration_wildlife_impact/styles/treespecstyle.qml zwar 14 Baumarten definiert sind aber nur 9 Symbole?

@twiebke übersehe ich hier etwas oder reichen die 9 Baumarten?

**Comment by twiebke** [comment]
Die reichen sicher nicht und ich würde da gerne auch noch weiter arbeiten. Bisher sind es nur die Baumartengruppen, weitestgehend nach Brandenburger Forsteinrichtungsanweisung. Kompatibilität wäre wünschenswert erfordert aber noch ein wenig Phantasie. Das können wir aber später nachführen, oder?

**Comment by b-lack** [comment]
Ja, das lässt sich beliebig erweitern. Es ist aber hardcoded und daher relative Zeitintensiv.

Ich habe in einem anderen Projekt einen Farbcode aus der ID der Baumart generiert. Das wäre ein effizienter Ansatz und garantiert, dass alle Farben einmalig sind.

**Comment by twiebke** [comment]
Lasse uns das so machen. Wichtiger ist die Eindeutigkeit und Einmaligkeit in der Darstellung und eine effiziente Umsetzung.
